### ISSUE TITLE
Hotbar use fails and causes character to not be able to move.

### ISSUE THREAD

**Issue by Randolphjand** [suriyun-production/mmorpg-kit-docs#1432]
I tried this on a fresh install of 2020 and 2022.  Put the fireball from items into hotbar slot and try to cast it.  It fails and then the character cannot move after that.  Hotbar works for things like food and water but not heal or fireball.  
NullReferenceException: Object reference not set to an instance of an object
MultiplayerARPG.DefaultCharacterUseSkillComponent.UseSkillItem (System.Int16 itemIndex, System.Boolean isLeftHand, System.UInt32 targetObjectId, MultiplayerARPG.AimPosition aimPosition) (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterSystems/CharacterActionsSystem/DefaultCharacterUseSkillComponent.cs:543) MultiplayerARPG.BaseCharacterEntity.UseSkillItem (System.Int16 itemIndex, System.Boolean isLeftHand, System.UInt32 targetObjectId, MultiplayerARPG.AimPosition aimPosition) (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterEntity/BaseCharacterEntity.cs:472)
MultiplayerARPG.PlayerCharacterController.RequestUsePendingSkill () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterControllerSystems/Default/PlayerCharacterController.cs:374)
MultiplayerARPG.PlayerCharacterController.UpdateQueuedSkill () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterControllerSystems/Default/PlayerCharacterController_Inputs.cs:569)
MultiplayerARPG.PlayerCharacterController.UpdateInput () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterControllerSystems/Default/PlayerCharacterController_Inputs.cs:157) 
MultiplayerARPG.PlayerCharacterController.Update () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterControllerSystems/Default/PlayerCharacterController.cs:229)

**Comment by insthync** [comment]
Fix it by add
```
            skill = item.UsingSkill;
            skillLevel = item.UsingSkillLevel;
``` 
to `CharacterDataExtension.cs` line `2007-2008` (in `ValidateSkillItemToUse` function)

**Comment by insthync** [comment]
And also add these codes

```
                // Decrease item immediately
                if (!Entity.DecreaseItemsByIndex(itemIndex, 1))
                {
                    return;
                }
                Entity.FillEmptySlots();
```

To `DefaultCharacterUseSkillComponent` class to fix item not being used (by host) issues

![Screenshot_934](https://user-images.githubusercontent.com/2697550/151074401-8cedffee-b9cc-4c83-9cce-da49f6c8823d.png)

**Comment by insthync** [comment]
Fixed in 1.73b
